src/style/style_layer/circle_style_layer.js

```diff
@@ -41,15 +44,56 @@ class CircleStyleLayer extends StyleLayer {
                            geometry: Array<Array<Point>>,
                            zoom: number,
                            bearing: number,
-                           pixelsToTileUnits: number): boolean {
+                           pixelsToTileUnits: number,
+                           cameraToCenterDistance: number,
+                           posMatrix: Float32Array,
+                           transform: Transform): boolean {
         const translatedPolygon = translate(queryGeometry,
             this.paint.get('circle-translate'),
             this.paint.get('circle-translate-anchor'),
             bearing, pixelsToTileUnits);
-        const radius = this.paint.get('circle-radius').evaluate(feature) * pixelsToTileUnits;
-        const stroke = this.paint.get('circle-stroke-width').evaluate(feature) * pixelsToTileUnits;
-        return multiPolygonIntersectsBufferedMultiPoint(translatedPolygon, geometry, radius + stroke);
+        const radius = this.paint.get('circle-radius').evaluate(feature);
```

Comment:
This all looks right to me but it took me a while to reason through it. Would this sort of explanatory comment help?

```
// For pitch-alignment: map, compare feature geometry to query geometry in the plane of the tile
// Otherwise, compare geometry in the plane of the viewport
// A circle with fixed scaling relative to the viewport gets larger in tile space as it moves into the distance
// A circle with fixed scaling relative to the map gets smaller in viewport space as it moves into the distance
```